backend/src/packages/chaiNNer_onnx/__init__.py

```diff
@@ -68,8 +76,68 @@ def get_onnx_runtime():
             size_estimate=2.5 * MB,
         ),
     ],
+    icon="ONNX",
+    color="#63B3ED",
 )
 
+if not is_arm_mac:
+    nv = get_nvidia_helper()
+    gpu_list = nv.list_gpus() if nv is not None else []
+
+    package.add_setting(
+        DropdownSetting(
+            label="ONNX GPU",
+            key="gpu",
+            description="Which GPU to use for ONNX. This is only relevant if you have multiple GPUs.",
+            options=[{"label": x, "value": str(i)} for i, x in enumerate(gpu_list)],
+            default="0",
+            disabled=not nvidia_is_available or len(gpu_list) <= 1,
+        )
+    )
+
+execution_providers = []
+try:
+    import onnxruntime as ort
+
+    execution_providers = ort.get_available_providers()
+except:
+    pass
```

Comment:
That's just not a good idea. What's the point of having these settings when they are functionally useless? 

I think we should only define settings when the package dependencies are met. The trick I used for features could be used here as well. We make a file called `package/settings.py` which adds the settings to the package and exports a `get_settings()` function. This function will be the only way to access settings, so nodes that need settings will import that function. So settings are only defined when nodes that use those settings are available. This also means that we can use all package dependencies in `settings.py`, so no optional imports.